docs/clusterobservability.md

```diff
@@ -166,19 +166,77 @@ The various storage configuration per capability can be achieved by multiple `Cl
 ### Object storage types
 
 Each object storage type has its own set of required fields which are configured in the secret.
-Object storage types will be added separately, there are plans to support the all the storage types supported by the capabilities.
+There are plans to support all storage types required by the capabilities.
 
-#### S3 secret
+#### Design principles
+
+The object storage configuration in COO follows the existing design principles of the Tempo and Loki operators.
+The `ClusterObservability` CR contains the type of the object storage and a secret holds the configuration.
+
+##### Alternative thanos-io/objstore
+
+The https://github.com/thanos-io/objstore client is used by Loki and Thanos to connect to the object storage.
+It offers a unified interface for different object storage types with a common configuration file.
+
+At the time of writing this document, the COO does not support Thanos, Loki operator does not directly support the `thanos-io/objstore`
+configuration file and the Tempo does not support it at all.
+
+#### Amazon S3 S3 / MinIO
+
+Supported by Tempo and Loki.
 
 ```yaml
-kubectl create secret generic s3 \
+kubectl create secret generic storage-secret \
     --from-literal=bucket="<BUCKET_NAME>" \
     --from-literal=endpoint="<AWS_BUCKET_ENDPOINT>" \
     --from-literal=access_key_id="<AWS_ACCESS_KEY_ID>" \
     --from-literal=access_key_secret="<AWS_ACCESS_KEY_SECRET>" \
     --from-literal=region="<AWS_REGION_YOUR_BUCKET_LIVES_IN>"
 ```
 
+* `region` - is optional in Tempo and required by Loki.
+
+##### Short lived - Amazon S3 with Security Token Service (STS)
+
+Supported by Tempo and Loki.
+
+```yaml
+kubectl create secret generic storage-secret \
+--from-literal=bucket="<BUCKET_NAME>" \
+--from-literal=role_arn="<AWS_ROLE_ARN>" \
+--from-literal=region="<AWS_REGION_YOUR_BUCKET_LIVES_IN>"
+```
```

Comment:
This does not really have something to do with how we specify the storage configuration, but I wonder why we always specify Secrets using these `kubectl create` commands instead of just providing a YAML example.